Add Avail Adapter for Sovereign SDK Integration and Examples

vibhurajeev opened the pull request:
# Description

## Overview
This pull request introduces the Avail Adapter to the Sovereign SDK. Additionally, examples demonstrating how to utilize this new adapter have been included.

## Changes Made
- Added the Avail Adapter to the Sovereign SDK.
- Included examples showcasing how to utilize the Avail Adapter.


## Documentation
The Avail Adapter and its integration into the Sovereign SDK are documented in the following locations:
- Updated the SDK documentation to include information about the new Avail Adapter and its usage.
- The examples repository contains detailed documentation on how to use the Avail Adapter for various use cases.

## Additional Changes
A minor enhancement was made to the sov-stf-runner:
- Separated da-config and rollup-config for improved clarity and maintainability.

vibhurajeev committed:
Avail adapter (#7)

* Add avail adapter.

* Add demo-rollup-avail.

* docs(da-service): Updated comments

* fix(demo-rollup-avail): Load client urls from toml

* docs(demo-rollup-avail): Updated docs to run avail node and light client.

* fix(demo-rollup-avail): Minor test fixes in config

* docs(demo-rollup-avail): fixed typos

* Updated adapter for new interface (#2)

* feat(avail-adapter): Updated as per new interface

* feat(avail-adapter): Added hash method

* fix(avail-adapter): code style

* fix(avail-adapter): minor code style

* Vr/demo rollup (#3)

* feat(avail-adapter): Updated as per new interface

* feat(avail-adapter): Added hash method

* fix(demo-rollup): Fix as per new interface

* fix(demo-rollup): rebase bugs

* Vr/update adapter (#5)

* feat(avail-adapter): Changes related to upstream update

* fix(avail-adapter): formatting

* fix(avail-adapter): cleanup

* Vr/prover (#4)

* feat(avail-adapter): Updated as per new interface

* feat(avail-adapter): Added hash method

* fix(demo-rollup): rebase bugs

* feat(demo-prover-avail): Prover demo with Avail

* feat(demo-prover-avail): modified adapter to work in zkvm

* fix(demo-prover-avail): outdated import

* feat(demo-rollup-avail): Makefile

* fix(demo-prover-avail): minor fixed

* fix(demo-prover-avail): minor fixes

* feat(demo-prover-avail): create private key

* fix(demo-prover-avail): read json file without jq

* fix(demo-prover-avail): typo

* fix(demo-prover-avail): GNU compatibility

* fix(demo-prover-avail): cleanup

* fix(demo-prover-avail): Removed logs

* fix(avail-adapter): cleanup

* fix(demo-rollup-avail): cleanup

* fix(demo-rollup-avail): update make file

* fix(avail-helper): formatting

* fix(demo-rollup-avail): update documentation (#6)

* fix(demo-rollup-avail): documentation fix

* fix(sov-cli): removed generated .dat files

* fix(sovereign-sdk): formatting

---------

Co-authored-by: Aleksandar Terentić <aleksandar@ethernal.tech>

vibhurajeev committed:
Vr/fix/rebase (#9)

* fix(avail-adapter): Errors introduced by rebase

* fix(demo-rollup-avail): Temp fixes

* fix(sov-stf-runner): Remove DA config

* fix(demo-rollup): cleanup

* fix(demo-prover-avail): Bugs introduced by rebase

* fix(demo-prover-avail): cleanup

* fix(sdk): remove cargo config

* fix(examples): config

vibhurajeev committed:
fix(examples): Updated config in celestia examples (#10)

aterentic-ethernal commented on adapters/avail/Cargo.toml:
```diff
@@ -0,0 +1,29 @@
+[package]
+name = "presence"
```
Do we want to rename it?

aterentic-ethernal commented on examples/demo-rollup-avail/src/ledger_rpc.rs:
```diff
@@ -0,0 +1,80 @@
+    pub fn extract_query_args<I: DeserializeOwned>(
+        params: jsonrpsee::types::Params,
+    ) -> Result<QueryArgs<I>, jsonrpsee::core::Error> {
+        if let Ok(args) = params.parse() {
+            return Ok(args);
+        }
+        let ids: Vec<I> = params.parse()?;
+        Ok(QueryArgs(ids, Default::default()))
```
It seems that this never extracts ids, since line 78 is never reached, is that an issue or just obsolete code?

aterentic-ethernal commented on examples/demo-rollup-avail/src/ledger_rpc.rs:
```diff
@@ -0,0 +1,80 @@
+    rpc: &mut RpcModule<LedgerDB>,
+) -> Result<(), jsonrpsee::core::Error> {
+    rpc.register_method("ledger_getHead", move |_, db| {
+        db.get_head::<B, T>().map_err(|e| e.into())
```
We can replace this and following `map_err` with `?` since `?` is doing implicite `into`

vibhurajeev committed:
fix(demo-rollup-avail): Config tests

vibhurajeev committed:
fix(demo-prover-avail): docs

vibhurajeev committed:
Merge branch 'nightly' into main

preston-evans98 commented on adapters/avail/src/verifier.rs:
```diff
@@ -0,0 +1,62 @@
+
+    // Verify that the given list of blob transactions is complete and correct.
+    // NOTE: Function return unit since application client already verifies application data.
+    fn verify_relevant_tx_list<SimpleHasher>(
+        &self,
+        block_header: &<Self::Spec as DaSpec>::BlockHeader,
+        _txs: &[<Self::Spec as DaSpec>::BlobTransaction],
+        _inclusion_proof: <Self::Spec as DaSpec>::InclusionMultiProof,
+        _completeness_proof: <Self::Spec as DaSpec>::CompletenessProof,
+    ) -> Result<Self::ValidityCondition, Self::Error> {
+        let validity_condition = ChainValidityCondition {
+            prev_hash: *block_header.prev_hash().inner(),
+            block_hash: *block_header.hash().inner(),
+        };
+
+        Ok(validity_condition)
+    }
```
@vibhurajeev, this implementation breaks soundness.

The only way for a rollup to work without a smart contract is to allow the end-user (light client) to verify that the rollup's fork choice rule was followed. If the verifier can't check this, then a malicious prover can create a *valid* proof which extends a non-canonical chain, ultimately causing honest light clients to accept the wrong rollup state. 

In this implementation, the SNARK doesn't commit to the transactions that it's processed. It commits to the block header, but since this function doesn't tie the actual bytes of data that the rollup processes back to the block header, that doesn't constrain the prover. Given a block header, the prover can pretend that the block contained whatever transactions he wants and the SNARK won't stop him. 

As we discussed offline, it's ok to push the work of verifying the transaction list outside of the SNARK - that's exactly what `ValidityConditions` were added for. But, the SNARK still has to commit to the actual set of blobs that it received from the prover, not just the alleged block header.

I think the correct way to implement this is roughly...

1. Add an additional field to the `ChainValidityCondition` for `tx_root`, and put the hash of the `_txs: &[<Self::Spec as DaSpec>::BlobTransaction]` argument in that field
2. Add a `check`er for the validity condition which downloads the data from a light client and enforces that it matches those nodes. 
3. Change the `combine` rule for validity conditions to ensure that when you check the validity condition for block N, you're also forced ot recursively check back to Genesis. (This is necessary since just because the prover gave the correct tx list in block N doesn't mean he gave the correct one in block N-1).

vibhurajeev committed:
Vr/data root (#11)

* Chain state module (#598)

* WIP: Add optimsitic sync

* WIP: implement chain state module

* WIP: Add attester incentives

* Can compile

* Does compile

* Start working on attester incentives

* Implement attestation processing

* Finish unbonding

* Improve events

* Rename sov-attester-incentives to optimistic-workflow

* Refactor

* Start working on apply_sync_data_blob

* Refactoring apply_tx_blob

* Refactoring files

* Rewriting unbonding

* Refactor attestation processing

* Refactor process challenge

* Refactoring

* Fixing compile issues

* Does compile

* Everything compiles

* Add 2 phase unbonding back

* Updating tests

* Changing bonding proof

* Changing bonding proof

* Fixing get_with_proof

* Fixing chain updates

* Change STF interface to work on slot level

* Removing Misbehavour hint from parameters
* Fixing prover
* Update docs

* Adding chain state

* Starting chain state

* Fix lint

* Fixing warnings

* Integrating slot hooks

* Populating todos

* Adding data generation for different modules

* Implementing automatic bank data generation

* Fixing stf

* Compiles

* Fixing chain-state test, adding EncodeCall trait

* Fixing lints

* Chain state testing

* Simple chain state module tests

* All tests pass

* Fixing lints

* Fixing comments PR

* Fixing comments PR

* nit fix

* Fixing PR comments

* Deleting borsh compat

* Adding back github and config files

* Specifying commit for jmt

* Update module-system/sov-state/src/storage.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fixing comments PR

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fixing error handling

* Fixing lints

* Fixing lints

* Fix lints

* Update rollup-interface/src/state_machine/mocks.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Addressing PR comments

* Including PR comments

* Update Cargo.toml

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/guest.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Adding back genesis and call

* Fix lints

* Lint

* Restoring default context

* Fixing imports

* Fixing prover compiles

* Fix prover compiles

* Fixing end slot hook

* Reorganizing integration tests

* Reorganizing integration tests

* Adding unit tests for chain-state

* Update chain state tests

* Update apply slot signature

* Fix demo-prover

* Update adapters/risc0/src/guest.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fix lints

* Format demo-prover

* Fix feature data generators

* Update examples/demo-stf/Cargo.toml

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/guest.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fixing pr comments

* Fix lint

* Removing call from chain-state because provided by blanket

---------

Co-authored-by: Preston Evans <preston.b.evans@gmail.com>
Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Refactoring buf reader (#593)

* WIP: Add optimsitic sync

* WIP: implement chain state module

* WIP: Add attester incentives

* Can compile

* Does compile

* Start working on attester incentives

* Implement attestation processing

* Finish unbonding

* Improve events

* Rename sov-attester-incentives to optimistic-workflow

* Refactor

* Start working on apply_sync_data_blob

* Refactoring apply_tx_blob

* Refactoring files

* Rewriting unbonding

* Refactor attestation processing

* Refactor process challenge

* Refactoring

* Fixing compile issues

* Does compile

* Everything compiles

* Add 2 phase unbonding back

* Updating tests

* Changing bonding proof

* Changing bonding proof

* Fixing get_with_proof

* Fixing chain updates

* Refactoring buf reader

* Change STF interface to work on slot level

* Removing Misbehavour hint from parameters
* Fixing prover
* Update docs

* Adding chain state

* Starting chain state

* Fix lint

* Fixing warnings

* Integrating slot hooks

* Populating todos

* Adding data generation for different modules

* Implementing automatic bank data generation

* Fixing stf

* Compiles

* Fixing chain-state test, adding EncodeCall trait

* Fixing lints

* Chain state testing

* Simple chain state module tests

* All tests pass

* Fixing lints

* Fixing comments PR

* Fixing comments PR

* nit fix

* Fixing PR comments

* Deleting borsh compat

* Adding back github and config files

* Specifying commit for jmt

* Update module-system/sov-state/src/storage.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fixing comments PR

* Applying PR comments

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fixing error handling

* Fixing lints

* Fixing lints

* Fix lints

* Refactor buf reader

* Update rollup-interface/src/state_machine/mocks.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Addressing PR comments

* Including PR comments

* Update Cargo.toml

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/guest.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Adding back genesis and call

* Fix lints

* Lint

* Restoring default context

* Fixing imports

* Fixing prover compiles

* Fix prover compiles

* Fixing end slot hook

* Reorganizing integration tests

* Reorganizing integration tests

* Adding unit tests for chain-state

* Update chain state tests

* Update apply slot signature

* Fix demo-prover

* Update adapters/risc0/src/guest.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fix lints

* Format demo-prover

* Fix feature data generators

* Merge with chain-state and fix lints

* Update examples/demo-stf/Cargo.toml

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/guest.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fixing pr comments

* Fix lint

* Fix lints

---------

Co-authored-by: Preston Evans <preston.b.evans@gmail.com>
Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Add missing docs for bank, prover-incentives, demo-rollup and const-rollup-config (#592)

* WIP: Add optimsitic sync

* WIP: implement chain state module

* WIP: Add attester incentives

* Can compile

* Does compile

* Start working on attester incentives

* Implement attestation processing

* Finish unbonding

* Improve events

* Rename sov-attester-incentives to optimistic-workflow

* Refactor

* Start working on apply_sync_data_blob

* Refactoring apply_tx_blob

* Refactoring files

* Rewriting unbonding

* Refactor attestation processing

* Refactor process challenge

* Refactoring

* Fixing compile issues

* Does compile

* Everything compiles

* Add 2 phase unbonding back

* Updating tests

* Changing bonding proof

* Changing bonding proof

* Fixing get_with_proof

* Fixing chain updates

* Missing docs bank

* Missing docs prover-incentives

* Missing docs const-rollup-config

* demo-rollup missing docs, finish missing docs

* Refactoring buf reader

* Fixing lint error

* Change STF interface to work on slot level

* Removing Misbehavour hint from parameters
* Fixing prover
* Update docs

* Adding chain state

* Starting chain state

* Fix lint

* Fixing warnings

* Integrating slot hooks

* Populating todos

* Adding data generation for different modules

* Implementing automatic bank data generation

* Fixing stf

* Compiles

* Fixing chain-state test, adding EncodeCall trait

* Fixing lints

* Chain state testing

* Simple chain state module tests

* All tests pass

* Fixing lints

* Fixing comments PR

* Fixing comments PR

* nit fix

* Fixing PR comments

* Deleting borsh compat

* Adding missing_docs

* Adding back github and config files

* Specifying commit for jmt

* Update module-system/sov-state/src/storage.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fixing comments PR

* Fix rpc macro

* Fixing missing docs

* Applying PR comments

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fixing error handling

* Fixing lints

* Fixing lints

* Fix lints

* Refactor buf reader

* Fixing tests

* Update rollup-interface/src/state_machine/mocks.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Addressing PR comments

* Including PR comments

* Update Cargo.toml

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/guest.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Adding back genesis and call

* Fix lints

* Lint

* Restoring default context

* Fixing imports

* Fixing prover compiles

* Fix prover compiles

* Fixing end slot hook

* Reorganizing integration tests

* Reorganizing integration tests

* Adding unit tests for chain-state

* Update chain state tests

* Update apply slot signature

* Fix demo-prover

* Update adapters/risc0/src/guest.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/host.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fix lints

* Format demo-prover

* Fix feature data generators

* Merge with chain-state and fix lints

* Update examples/demo-stf/Cargo.toml

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Update adapters/risc0/src/guest.rs

Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* Fixing pr comments

* Fix lint

* Fix doctests demo rollup

* Add folder generated by README test to gitignore

---------

Co-authored-by: Preston Evans <preston.b.evans@gmail.com>
Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>

* EVM: Recovering transaction signer in the EVM module. (#639)

* Refactor `CelestiaAddress` (#551)

* Refactor CelestiaAddress and its internal representation

* Correct comparison for sender (#660)

* Remove `associated_type_defaults` making Sovereign SDK stable compatible (#649)

* Based Sequencer with Soft Confirmations: deferring blob from execution (#596)

* Introducing blob deferring
* Exposing is_allowed_sequencer method. Filter non registered sequencers
* Add next-test as default runner for `make test`

* Allow custom (de)serialization logic for `#[state]` (#648)

* Allow custom state encodings

* Use Default::default to build codecs

* Improve tests for sov-modules-macros

* Stricter type bounds for better error messages

* Don't reexport sov_state::codec

* t.pass test

* Bump tokio from 1.29.1 to 1.31.0 (#659)

Bumps [tokio](https://github.com/tokio-rs/tokio) from 1.29.1 to 1.31.0.
- [Release notes](https://github.com/tokio-rs/tokio/releases)
- [Commits](https://github.com/tokio-rs/tokio/compare/tokio-1.29.1...tokio-1.31.0)

---
updated-dependencies:
- dependency-name: tokio
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Filippo Neysofu Costa <filippo@sovlabs.io>

* Add Demo NFT module docs (#642)

* Make Sequencer Outcome generic over address (#652)

* Mock refactoring in `rollup-interface/src/state_machine/mocks` (#661)

* Add a benchmark/profiler to count cycles in demo-prover (#577)

* extract data_gen into its own function

* temp push

* add prover cycles

* include multiple transactions

* reduce txns

* temp switch

* cycle tracker macro

* another temp commit

* merge main and nightly

* use run_without_prover for bench/profile

* working metrics

* add zk-cycle-utils. missed it

* temporary commit

* pass features to risc0 vm

* feature gate all the bench things

* cleanup and testing

* check in readme

* checking some cargo locks in prep for merge

* some fixes

* revert rollup config toml

* working merge

* cleanup and documentation

* simplify the macro for generating the wrapped function

* format fixes

* README changes

* simplify code as per lint

* lint fix

* add profiler to analyze trace

* fix cargo lock

* really fix Cargo.lock

* trace options

* cargo fmt

* fix log error

* fix prover bench

* add docs and address comments

* prover bench readme

* add a skip proving option to demo-prover

* add tests for macros

* demo prover format

* remove commented code

* formatting

---------

Co-authored-by: dubbelosix <dub@006.com>

* Simplify generics in `AppTemplate`.  (#662)

* Move module-template tests into dir (#665)

* Use DaService in `sov-ethereum`. (#668)

* Rename jupiter to celestia (#670)

* Config preferred sequencer via bool (#672)

* `sov-sequencer-registry` module docs (#674)

* Fix sov-sequencer-registry docs

* Fix docs about callmessage

* `RollupConfig` cleanup (#677)

* Typo in ledger_rpc.rs (#678)

* fixed typo in ledger_rpc.rs

* Fix RPC method name in sov-stf-runner docs

---------

Co-authored-by: Filippo Costa <filippo@sovlabs.io>

* Rollup strcture in demo-rollup (#680)

* Optimistic (#650)

* Optimistic

* Attester incentives module

* Reverting typo

* Tests attester-incentives

* Adding test helper and lint

* Lint

* Adding tests

* Finishing positive test

* Refactor tests

* Refactor tests

* Adding negative test

* Valid challenge

* Invalid challenge

* Adding unbonding tests

* Lint fix

* Lint

* Make lint

* Make lint

* PR comments + fix tests

* Lint

* Fixing comments

* Refactoring + finishing adding tests

* Fix lint

* Fix lint

* Fixing PR comments

* Fixing important comment

* Fix nits

* Fixing multi-attestation

* Fixing nits

* Fixing nits

* Fixing lints

* Fixing all comments PR

* Celestia docker compose: several nodes (#682)

* Bump parking_lot from 0.11.2 to 0.12.1 (#696)

Bumps [parking_lot](https://github.com/Amanieu/parking_lot) from 0.11.2 to 0.12.1.
- [Changelog](https://github.com/Amanieu/parking_lot/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Amanieu/parking_lot/compare/0.11.2...0.12.1)

---
updated-dependencies:
- dependency-name: parking_lot
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump serde from 1.0.183 to 1.0.185 (#698)

Bumps [serde](https://github.com/serde-rs/serde) from 1.0.183 to 1.0.185.
- [Release notes](https://github.com/serde-rs/serde/releases)
- [Commits](https://github.com/serde-rs/serde/compare/v1.0.183...v1.0.185)

---
updated-dependencies:
- dependency-name: serde
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump quote from 1.0.32 to 1.0.33 (#701)

Bumps [quote](https://github.com/dtolnay/quote) from 1.0.32 to 1.0.33.
- [Release notes](https://github.com/dtolnay/quote/releases)
- [Commits](https://github.com/dtolnay/quote/compare/1.0.32...1.0.33)

---
updated-dependencies:
- dependency-name: quote
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump tokio from 1.31.0 to 1.32.0 (#702)

Bumps [tokio](https://github.com/tokio-rs/tokio) from 1.31.0 to 1.32.0.
- [Release notes](https://github.com/tokio-rs/tokio/releases)
- [Commits](https://github.com/tokio-rs/tokio/compare/tokio-1.31.0...tokio-1.32.0)

---
updated-dependencies:
- dependency-name: tokio
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Filippo Neysofu Costa <filippo@sovlabs.io>

* Run CI even for *.md changes (#704)

* Fix typos in documentation (#687)

* fix typos

* fix typos

* fix typo

* fix typos

* fix typo

* fix typo

* fix typo

* fix typo

* fix typo

* fix typo

* fix typo

* Update examples/demo-rollup/benches/README.md

Fix typo

* Trigger CI

---------

Co-authored-by: Filippo Neysofu Costa <filippo@neysofu.me>
Co-authored-by: Filippo Costa <filippo@sovlabs.io>

* Update README.md (#683)

* Update README.md

To make getting started section more digestable.

* Update README.md

---------

Co-authored-by: Filippo Neysofu Costa <filippo@sovlabs.io>

* `PairOfCodecs` default (#703)

* Make "native" feature non default (#664)

* Disabling default native feature

* Fixes after rebase

* Functionality for submitting several blobs in single DA block (#690)

* Introduce sciprts for testing deferred blob execution

* Clean up logging
* Scripts for make demo
* Improve logging
* Use preferred sequencer
* Log transaction hash in batch builder

* Refactor `sov-cli` into a library (#708)

* WIP: add rpc to cli wallet

* Test key commands

* WIP allow generic callmsg parsing

* It kinda works

* Switch to tryfrom

* It works! needs cleanup

* It works a bit better

* Use new frontend trait

* Import from files and json

* Test import flows

* Rename to fileNameArg

* Implement tx submission via rpc

* Update sov-cli in demo-stf

* Cleanup diff

* Cleanup diff

* Remove unneeded json file

* lint

* remove sov-cli from guest

* lint

* Fix tests

* Remove outdated clap test

* Fix sequencer tests

* Add build step to makefile

* Remove stray ':' from Makefile

* Update test for cli_wallet derive

* DA mock implementation (#686)

* Minor fixes in demo-rollup to filter out transaction registrations (#712)

* Implement State Vectors (#685)

* impl. StateVec

* remove remove func. impl. set_all

* fix set_all

* add tests

* fix naming

* borrow values

* impl. get_or_err

* example without tests yet

* formatting

* docs: document sov-vec-setter

* lint: sov-vec-setter

* fix fn order

* string test case

* linting

* Add tests for sov-vec-setter and StateVec

* Update doc for set_all

* Fix Clippy warnings

* Fix StateVec::clear() bug

---------

Co-authored-by: orkunkilic <orkunmahirkilic@gmail.com>
Co-authored-by: Filippo Costa <filippo@sovlabs.io>

* Improvements to `StateVec` (#688)

Signed-off-by: Filippo Costa <filippo@sovlabs.io>

* Hash-encode keys in `StateMap` instead of serializing them with a codec (#718)

* Use Hash to serialize keys

* Better naming conventions for Storage/Cachekey

* Fix docs

* Fix merge issues

Signed-off-by: Filippo Costa <filippo@sovlabs.io>

---------

Signed-off-by: Filippo Costa <filippo@sovlabs.io>

* Split AddressTrait (#720)

* Removing extra bound on RollupAddress where it is not needed
* Update demo-simple-stf docs
* Public export
* Remove unused H160

* EVM remove receipt type (#719)

* Use `MockAddress` in `sov-sequencer-registry` tests.  (#725)

* Fix `#[state(codec_builder = ...)]` test (#723)

Signed-off-by: Filippo Costa <filippo@sovlabs.io>

* Add create-token integration test (#716)

* Use update proofs

* Add basic module integration tests

* Randomize port in tests

* Revert "Use update proofs"

This reverts commit e5bfaca76db67d8aaa5fbc1436da20c8c39d3de6.

* address code review; reorganize

* Add ws subscription for slots

* fix lints

* Fix bash snippet in README interpreted as test

* Fixing `where` bounds for rpc_gen (#726)

* Generate docs in rpc macros (#729)

* Add missing docs to generated rpc

* Remove allow(missing_docs) from query modules

* remove println

* add missing docs for attester-incentives

* EVM test cleanup (#732)

* Fix CI (#734)

* Add docs for examples/demo-prover (#730)

Co-authored-by: dubbelosix <dub@006.com>

* Use MockAddress in demo-rollup benches. (#738)

* Fix Blob address type in benches

* fix lint

* Bump serde from 1.0.185 to 1.0.188 (#746)

Bumps [serde](https://github.com/serde-rs/serde) from 1.0.185 to 1.0.188.
- [Release notes](https://github.com/serde-rs/serde/releases)
- [Commits](https://github.com/serde-rs/serde/compare/v1.0.185...v1.0.188)

---
updated-dependencies:
- dependency-name: serde
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump reqwest from 0.11.18 to 0.11.20 (#745)

Bumps [reqwest](https://github.com/seanmonstar/reqwest) from 0.11.18 to 0.11.20.
- [Release notes](https://github.com/seanmonstar/reqwest/releases)
- [Changelog](https://github.com/seanmonstar/reqwest/blob/master/CHANGELOG.md)
- [Commits](https://github.com/seanmonstar/reqwest/compare/v0.11.18...v0.11.20)

---
updated-dependencies:
- dependency-name: reqwest
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* CLEANUP: remove sov-election module since we don't need it and modify tests using it to use sov-bank (#739)

* cleanup sov election

* fix unused import

* fix removed arguments to genesis config

* remove extra argument for create_demo_genesis_config

* add a bad nonce test case as well

* formatting :(

* another place where private key for election module needs to be removed

* linter suggestions

---------

Co-authored-by: dubbelosix <dub@006.com>

* Update `end_slot_hook` (#742)

* add root_hash to end_sloot_hook

* update end slot hook

* support experimental

* allow unused variables

* fix experimental flag for hooks in evm

* Migrate to buildjet (#748)

* Migrate to buildjet

* Switch back to swatinem cache

* No default feature in CI (#741)

Signed-off-by: Filippo Costa <filippo@sovlabs.io>

* Allow modules to mint on their own authority (#728)

* Allow modules to mint on their own authority

* Fix typos in token::mint signature

* Allow modules to create tokens

* Fix nits

* lint

* Fix tests

* Fix tests

* Modify risc0 tracer util to print function stack to enable better debugging (#711)

* stack analysis

* fix issues

* some more changes

* minor changes

---------

Co-authored-by: dubbelosix <dub@006.com>

* Remove  unnecessary generic from `MockBlob` (#749)

* Remove generic from MockBlob

* update tests

* fix lint

* fix lint

* feat: add fuzzing (#651)

* macro docs improvements

* wip

* fuzz experiments

* macro errors

* wip

* add fuzz makefile & docs

* add generative fuzz for account module implementation

This commit introduces a generative model for the accounts module
implementation that will generate valid messages to update the account
address.

It also fixes a bug where a prefix with an unaligned vector could be
created by declaring an API module, combining its prefixes, and
generating a vec would result in potentially unaligned vectors.

* fix statemap arbitrary generation

* fix lint nits

* rollback aligned vec assertion for tests

* fix native requirement for arbitrary implementation

* fix arbitrary workset to use genesis

* fix merge generated errors

* remove cargo.lock from fuzz directory

---------

Co-authored-by: Filippo Costa <filippo@sovlabs.io>

* Fix CI job cancellation (#751)

* Split `Storage::validate_and_commit` into two separate methods `Storage::calculate_root_hash...` and `Storage::commit` (#752)

* Extend storage interface

* fix

* add calculate_state_root_and_node_batch

* Add comment

* code review feedback

* Add StateUpdate to Storage

* Fix `expose_rpc` macro with associated types (#733)

* Start adding test

* Add simple test for expose_rpc

* Generic test passes without expose_rpc

* It still passes.

* Just adding some notes

* Add test boilerplate

* [no ci]: Update tests

* Broken, but progress

* [no ci] It passes!

* And add failing test!

* Adding module for debugging macro things

* It compiles and runs!

* Adding generic!

* Add phantom module to satisfy compiler

* Fixing phantom

* WIP

* It compiles without expose_rpc

* We don't need phantom module

* Update things

* It works with renaming!

* Linter is happy

* tests are passing

* Add failing test

* Clean ups

* Add link to bug

* Update message

* Allow modules to burn coins (#756)

* Move generic from method to trait, so it will be easier to use it with DA spec (#759)

* Add Bitcoin DA Adapter link to README (#758)

* docs: add Bitcoin DA link to README

* docs: add "open-source" to BitcoinDA link

Co-authored-by: Cem Özer <cemozer2018@u.northwestern.edu>

---------

Co-authored-by: Cem Özer <cemozer2018@u.northwestern.edu>

* Add RPC for `chain-state` module (#760)

* Add RPC endpoint for chain state module, so it can be plugged into runtime

* Move tests to integration tests

* Fix native feature gating

* Fixed check features

* Specify version of `jmt` which contains fix (#762)

* Specify version of `jmt` which contains fix

* Fix jsonrpsee new method name

* Bump prover's cargo lock

* Remove separate "serde" feature (#761)

* Remove separate "serde" feature

As it is used only inside native feature and not separately

* Fix native test for sov-attester-incentives

* Bump dalek-ed25519 to 2.0.0 (#667)

* Bump dalek-ed25519 to 2.0.0

* Fix serde serialization

* Fixing lint and comments

* Adding key converters and helpers

* Remove temporary HexKey

* Update key files structure

* Fixing test!

* Addressing nits

* Fix `expose_rpc` macro to handle all cases of generics from Runtime (#764)

* Start working on it

* Use different approach with phantom data, but...

Lifetime strikes back so far

* Start cleaning up and adding tests

* Adding more tests and cleaning up

* Adding some tests instead of removed ones

* Fix lint error

* Actually fix lint

* EVM: Allow DA blobs with multiple transactions. (#767)

* ETH batch builder

* EVM: allow building blocks with multiple transactions

* undo changes in full-node/sov-sequencer/src/utils.rs

* Add comments in tests

* fix lint

* fix cargo hack

* Update to HexPrivateAndAddress

* Add doc generation to the `genesis` macro. (#772)

* Add doc generation to genesis macro

* make fields public

* fix lint

* Add a test case to `demo-simple-stf` (#774)

* add test case to demo-simple-stf

* renaming

* Add docs to dispatch call macro (#775)

* Plug in `sov-chain-state` module (#776)

* Introduce Da generic in demo-stf

* Move sov-cli to demo-rollup

* Clean up sov-cli usage

* Refactor Generics to use Specs (#780)

* s/DA/Da/g

* Progress

* Progressing

* Continue grouping generics

* Adjust BlobSelector

* Update module-system/module-implementations/sov-attester-incentives/src/call.rs

* fix(examples): rebase

* fix(avail): tx commitment

* feat(demo-rollup): new_rollup_with_avail_da

* fix(demo-prover-avail): rebase fixes

* fix(demo-prover-avail): cleanup and import changes

* fix(demo-prover-avail): fixed clap versioning issue

* fix(avail-adapter): blake hasher target

* feat(avail-adapter): send_transaction

---------

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Filippo Costa <filippo@sovlabs.io>
Co-authored-by: theochap <80177219+theochap@users.noreply.github.com>
Co-authored-by: Preston Evans <preston.b.evans@gmail.com>
Co-authored-by: Nikolai Golub <nikolai@sovlabs.io>
Co-authored-by: Blazej Kolad <blazejkolad@gmail.com>
Co-authored-by: Filippo Neysofu Costa <filippo@sovlabs.io>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: dubbelosix <rohan@sovlabs.io>
Co-authored-by: dubbelosix <dub@006.com>
Co-authored-by: Andreas Tzionis <Andy0n7@gmail.com>
Co-authored-by: omahs <73983677+omahs@users.noreply.github.com>
Co-authored-by: Filippo Neysofu Costa <filippo@neysofu.me>
Co-authored-by: Cem Özer <cem@sovlabs.io>
Co-authored-by: Preston Evans <32944016+preston-evans98@users.noreply.github.com>
Co-authored-by: Esad Yusuf Atik <esadyusufatik@gmail.com>
Co-authored-by: orkunkilic <orkunmahirkilic@gmail.com>
Co-authored-by: Victor Lopes <victor@codx.io>
Co-authored-by: Cem Özer <cemozer2018@u.northwestern.edu>

vibhurajeev committed:
Merge branch 'nightly'

vibhurajeev committed:
fix(avail-adapter): rebase changes

vibhurajeev committed:
fix(avail-adapter): rebase introduced bugs

vibhurajeev committed:
fix(avail-adapter): cleanup

preston-evans98 commented:
@vibhurajeev, finally had a chance to start working on this! I've PR'd the first chunk (the adapter itself as #905). It needed a few modifications to pass CI, so I've opened a PR against your fork to show what the changes are. No need to merge that PR - just give an approval on #905 if it looks good to you.

cemozerr commented:
Given that #905 is merged, can we close this PR? @preston-evans98

vibhurajeev committed:
Merge branch 'nightly'

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Sovereign-Labs/sovereign-sdk?pullRequest=631) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Sovereign-Labs/sovereign-sdk?pullRequest=631) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Sovereign-Labs/sovereign-sdk?pullRequest=631) it.</sub>
